### PR TITLE
update: payment type enum added

### PR DESCRIPTION
**Pull request by hvsharma63**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     ],
     "require": {
         "php": "^8.2",
+        "bensampo/laravel-enum": "^6.7",
```
Comment (gauravmak) [inline composer.json:25]: This is not needed. PHP supports enums out of the box.

https://github.com/BenSampo/laravel-enum#migrate-to-native-php-enums

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: src/Enums/PaymentType.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    case AMEX = 'amex';
+
+    case VOUCHER = 'voucher';
+
+    case OTHERS = 'others';
+
+    public static function names(): array
+    {
+        return array_column(self::cases(), 'name');
+    }
```
Comment (gauravmak) [inline src/Enums/PaymentType.php:24]: Are we using this?

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: src/Enums/PaymentType.php (hunk 2 above)
Comment (hvsharma63) [reply src/Enums/PaymentType.php:24]: Nope. But added, thought it could be useful. being used in `array()`

Comment (gauravmak) [reply src/Enums/PaymentType.php:24]: We can remove both for now and add when needed please

**Comment by gauravmak** [comment]
@hvsharma63 Please update `tests/Datasets/SalesData.php` and also add a validation that the payment keys have to be one of the enum values.

File: src/Commands/SalesFileGenerationCommand.php [hunk 3]
```diff
@@ -226,4 +229,35 @@ private function generateSalesFiles(array $config, Collection $stores, string $d
...
+            '*.discount' => ['required', 'decimal:0,2'],
+            '*.SST' => ['required', 'decimal:0,2'],
+            '*.payments' => ['required', "array:{$paymentTypes}"],
+            '*.payments.cash' => ['required', 'decimal:0,2'],
+            '*.payments.tng' => ['required', 'decimal:0,2'],
+            '*.payments.visa' => ['required', 'decimal:0,2'],
+            '*.payments.mastercard' => ['required', 'decimal:0,2'],
+            '*.payments.amex' => ['required', 'decimal:0,2'],
+            '*.payments.voucher' => ['required', 'decimal:0,2'],
+            '*.payments.others' => ['required', 'decimal:0,2'],
```
Comment (gauravmak) [inline src/Commands/SalesFileGenerationCommand.php:250]: Can we go via enum instead of direct string?

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: src/Commands/SalesFileGenerationCommand.php (hunk 3 above)
Comment (hvsharma63) [reply src/Commands/SalesFileGenerationCommand.php]: Yes. Made it dynamic.

File: tests/Datasets/SalesData.php [hunk 4]
```diff
@@ -48,7 +48,7 @@ function sampleSalesData2(): array
 {
     return [
         [
```
Comment (gauravmak) [inline tests/Datasets/SalesData.php:50]: Please use enum in the payment array

**Review by gauravmak — CHANGES_REQUESTED** [review]

**Pull request merged**